packages/snaps-sdk/src/jsx/validation.ts

```diff
@@ -0,0 +1,331 @@
+import {
+  assertStruct,
+  hasProperty,
+  isPlainObject,
+  JsonStruct,
+  StrictHexStruct,
+} from '@metamask/utils';
+import type { Struct } from 'superstruct';
+import {
+  is,
+  boolean,
+  defaulted,
+  optional,
+  tuple,
+  array,
+  lazy,
+  nullable,
+  number,
+  object,
+  record,
+  string,
+} from 'superstruct';
+import type { ObjectSchema } from 'superstruct/dist/utils';
+
+import type { Describe } from '../internals';
+import { literal, nullUnion } from '../internals';
+import type { EmptyObject } from '../types';
+import type {
+  GenericSnapElement,
+  Key,
+  MaybeArray,
+  StringElement,
+} from './component';
+import type {
+  AddressElement,
+  BoldElement,
+  BoxElement,
+  ButtonElement,
+  CopyableElement,
+  DividerElement,
+  FieldElement,
+  FormElement,
+  HeadingElement,
+  ImageElement,
+  InputElement,
+  ItalicElement,
+  JSXElement,
+  LinkElement,
+  RowElement,
+  SpinnerElement,
+  StandardFormattingElement,
+  TextElement,
+} from './components';
+
+/**
+ * A struct for the {@link Key} type.
+ */
+export const KeyStruct: Describe<Key> = nullUnion([string(), number()]);
+
+/**
+ * A struct for the {@link StringElement} type.
+ */
+export const StringElementStruct: Describe<StringElement> = maybeArray(
+  string(),
+);
+
+/**
+ * A struct for the {@link GenericSnapElement} type.
+ */
+export const ElementStruct: Describe<GenericSnapElement> = object({
+  type: string(),
+  props: record(string(), JsonStruct),
+  key: nullable(KeyStruct),
+});
+
+/**
+ * A helper function for creating a struct for a {@link MaybeArray} type.
+ *
+ * @param struct - The struct for the maybe array type.
+ * @returns The struct for the maybe array type.
+ */
+function maybeArray<Type, Schema>(
+  struct: Struct<Type, Schema>,
+): Struct<MaybeArray<Type>, any> {
+  return nullUnion([struct, array(struct)]);
+}
+
+/**
+ * A helper function for creating a struct for a JSX element.
+ *
+ * @param name - The name of the element.
+ * @param props - The props of the element.
+ * @returns The struct for the element.
+ */
+function element<Name extends string, Props extends ObjectSchema = EmptyObject>(
+  name: Name,
+  props: Props = {} as Props,
+) {
+  return object({
+    type: literal(name) as unknown as Struct<Name, Name>,
+    props: object(props),
+    key: nullable(KeyStruct),
+  });
+}
+
+/**
+ * A struct for the {@link ButtonElement} type.
+ */
+export const ButtonStruct: Describe<ButtonElement> = element('Button', {
+  children: StringElementStruct,
+  name: optional(string()),
+  type: defaulted(
+    optional(nullUnion([literal('button'), literal('submit')])),
+    'button',
+  ),
+  variant: defaulted(
+    optional(nullUnion([literal('primary'), literal('destructive')])),
+    'primary',
+  ),
+  disabled: defaulted(optional(boolean()), 'false'),
```

Comment:
```suggestion
  disabled: defaulted(optional(boolean()), false),
```
I assume this is wrong?